src/Core/Domain/Exception/DomainException.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Exception;
+
+/**
+ * Used when a domain constraint has been violated
+ */
+class DomainException extends \Exception
```

Comment:
Shouldn't it extend `RuntimeException` instead of `Exception` ?

Reply:
It _could_... but _why_? Would they be handled differently?

Reply:
Yes, according to the docs, the exceptions class hierarchy has 2 families :
* [`RuntimeException`](http://php.net/manual/en/class.runtimeexception.php)
* [`LogicException`](http://php.net/manual/en/class.logicexception.php)

The former for qualifying errors occurring on pre- and post-conditions, the latter on the code logic itself.

The `Exception` class comes from the Zend Engine which is good if you are writing a PHP extension but does not bring much information on its nature, the 2 others comes from the SPL and is designed for userland.